// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Internal wrapper proto for RPC call between host process and sandbox.
syntax = "proto3";
package google.scp.roma.proto;

import "src/roma/interface/function_binding_io.proto";

message RpcWrapper {
  // Proto containing input/output args for native function.
  FunctionBindingIoProto io_proto = 1;

  // The id of the invocation request associated with this RPC.
  // This id is client-set, so no guarantee of uniqueness.
  string request_id = 2;

  // Internal uuid associated with an invocation request.
  // Allows for retrieval of metadata based on invocation request uuid in the host process.
  string request_uuid = 3;

  // Name of native function to be invoked in host process
  string function_name = 4;
}
